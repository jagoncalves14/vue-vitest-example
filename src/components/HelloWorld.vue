<template>
  <div class="title">Hello World</div>
  <Bye />
</template>

<script>
import Bye from './Bye.vue';

export default {
  data() {
    return {
      hasMounted: false,
    };
  },

  components: {
    Bye,
  },

  mounted() {
    this.hasMounted = true;
  },
};
</script>
