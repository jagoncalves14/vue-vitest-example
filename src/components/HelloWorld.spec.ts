// @ts-ignore
import HelloWorld from 'components/HelloWorld.vue';
import { mount } from '@vue/test-utils';

describe('Component: HelloWorld', () => {
  it('Mounts properly', () => {
    const wrapper = mount(HelloWorld);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Mounts properly - shallow mount', () => {
    const wrapper = mount(HelloWorld, {
      shallow: true,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('mounted - updates hasMounted value', () => {
    const context = {
      hasMounted: false,
    };

    HelloWorld.mounted.call(context);

    expect(context.hasMounted).toBe(true);
  });
});
