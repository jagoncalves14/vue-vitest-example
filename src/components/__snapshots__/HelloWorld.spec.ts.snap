// Vitest Snapshot v1

exports[`Component: HelloWorld > Mounts properly - shallow mount 1`] = `
"<div class=\\"title\\">Hello World</div>
<bye-stub></bye-stub>"
`;

exports[`Component: HelloWorld > Mounts properly 1`] = `
"<div class=\\"title\\">Hello World</div>
<div class=\\"title\\">Bye</div>"
`;
